### PR TITLE
Motd UI

### PR DESCRIPTION
**Pull request by Fidaman**

### CODE REVIEW

File: motd/src/ts/MotD.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  constructor(props: MotdProps) {
+    super( props );
+  }
+
+  closeWindow(): void {
+    client.CloseUI( 'motd' );
+  }
+
+  onMessage = (data: MotdData) => {
+    if(data.message=='')
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: https://github.com/airbnb/javascript#whitespace--around-keywords

File: motd/src/ts/MotD.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  componentWillMount() {
+    this.setState( { message: '' } );
+    restAPI.getMessageOfTheDay().then( this.onMessage, this.onMessageFailed );
+  }
+
+  render() {
+    let messageContent: any;
+    if ( this.state.message ) {
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: as above

File: motd/src/ts/MotD.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    this.setState( { message: data.message } );
+    var timeout = data.duration * 1000;
+    setTimeout(() => { this.setState( { message: '' } ); }, timeout );
+    setTimeout(() => { this.closeWindow(); }, timeout + this.fadeTime );
+  }
+
+  onMessageFailed = (error: any) => {
+    console.log( error );
+      this.setState( { message: '' } );
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: indentation and whitespace--in-parens

``` JavaScript
this.setState({ message: '' });
```

https://github.com/airbnb/javascript#whitespace--in-parens
https://github.com/airbnb/javascript#whitespace--in-braces

File: motd/src/ts/MotD.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  componentWillMount() {
+    this.setState( { message: '' } );
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: as above

File: motd/src/ts/MotD.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  message: string;
+  duration: number;
+}
+
+class Motd extends React.Component<MotdProps, MotdState> {
+
+  private fadeTime:number = 500;
+
+  constructor(props: MotdProps) {
+    super( props );
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]:
``` JavaScript
super(props)
```

https://github.com/airbnb/javascript#whitespace--in-parens

File: motd/src/ts/MotD.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  closeWindow(): void {
+    client.CloseUI( 'motd' );
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]:
```JavaScript
client.CloseUI('motd');
```

File: motd/src/ts/MotD.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  onMessage = (data: MotdData) => {
+    if(data.message=='')
+      return;
+
+    console.log( "onMessage="+data.message );
```
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: whitespace--in-parens

File: motd/src/ts/MotD.tsx (hunk 6 above)
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: whitespace--in-parens

File: motd/src/ts/MotD.tsx (hunk 7 above)
Comment (Mehuge) [inline motd/src/ts/MotD.tsx]: whitespace--in-parens x 3

File: motd/jsconfig.json [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "compilerOptions": {
+    "target": "ES6"
+  }
+}
```
Comment (Mehuge) [inline motd/jsconfig.json]: missing newline at end of file

**Comment by Mehuge** [comment]
Looks good

File: motd/package.json [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  "private": true,
+  "name": "cu-ui-motd",
+  "version": "0.2.0",
+  "description": "Camelot Unchained UI - motd",
+  "license": "MPL-2.0",
+  "main": "src/ts/main.tsx",
+  "author": "City State Entertainment, LLC <support@citystateentertainment.com>",
+  "contributors": [
+    "saddieeiddas <saddie.eiddas@gmail.com>",
+    "JB (codecorsair) <jb@codecorsair.com>"
```
Comment (codecorsair) [inline motd/package.json]: You're a contributor on this module, not either of us two.  :)

**Comment by Fidaman** [comment]
ok my name is on this one now

**Comment by codecorsair** [comment]
Looks good. :+1:

**Pull request merged**